### PR TITLE
q5 reverse

### PR DESCRIPTION
**Pull request by wave49192**

### CODE REVIEW

File: reverse.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+def reverse(word)
+  puts word.reverse
+end
```
Comment (WatcharitNew) [inline reverse.rb:3]: What if I put `nil` here? Can you handle this case?

File: reverse.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+def reverse(word)
```
Comment (siripoom-datawow) [inline reverse.rb:1]: what if we not use built in method?
lets try another solution krubbb.
this will make you get used to handling with array.

**Review by siripoom-datawow — APPROVED** [review]

File: spec/reverse_spec.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+require './reverse.rb'
+
+RSpec.describe 'reverse' do
+  it "should return reversed string" do
+    expect(reverse('hello')).to eq('olleh')
+    expect(reverse('world')).to eq('dlrow')
+  end
+  it "should raise error when string is nil" do
```
Comment (WatcharitNew) [inline spec/reverse_spec.rb:8]: Add empty line to separate test

File: spec/reverse_spec.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    expect {reverse(nil)}.to raise_error(ArgumentError,"Word cannot be nil")
```
Comment (WatcharitNew) [inline spec/reverse_spec.rb:9]: Add empty space between `{}`

File: reverse.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  new_reversed_char = []
+  raise ArgumentError, "Word cannot be nil" if word.nil?
+  (word.length - 1).downto(0) do |i|
+    new_reversed_char << word[i]
+  end
+  return new_reversed_char.join('')
```
Comment (WatcharitNew) [inline reverse.rb:7]: You can delete `return` here since it is last line

File: reverse.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+def reverse(word)
+  new_reversed_char = []
+  raise ArgumentError, "Word cannot be nil" if word.nil?
+  (word.length - 1).downto(0) do |i|
```
Comment (WatcharitNew) [inline reverse.rb:4]: Add empty space after raising error

**Review by WatcharitNew — APPROVED** [review]

File: reverse.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+def reverse(word)
+  new_reversed_char = []
+  raise ArgumentError, "Word cannot be nil" if word.nil?
```
Comment (numin123) [inline reverse.rb:3]: Maybe check only nil case is not enough.
you can use `respond_to?(:length)` to check this obj is support this method or not.

**Pull request merged**